apps/guardian-ui/src/components/BalanceCard.tsx

```diff
@@ -1,21 +1,37 @@
-import { Card, CardBody, CardHeader, Text } from '@chakra-ui/react';
+import { Card, CardBody, CardHeader, Skeleton, Text } from '@chakra-ui/react';
 import { KeyValues } from '@fedimint/ui';
-import { useTranslation } from '@fedimint/utils';
-import React, { useMemo } from 'react';
+import { MSats, formatMsatsToBtc, useTranslation } from '@fedimint/utils';
+import React, { useEffect, useMemo, useState } from 'react';
+import { AuditSummary } from '../types';
+import { useAdminContext } from '../hooks';
 
 export const BalanceCard: React.FC = () => {
   const { t } = useTranslation();
+  const { api } = useAdminContext();
+  const [auditSummary, setAuditSummary] = useState<AuditSummary>();
+
+  const walletBalance = auditSummary
+    ? auditSummary.module_summaries.wallet?.net_assets || (0 as MSats)
+    : undefined;
+
+  useEffect(() => {
+    api.audit().then(setAuditSummary).catch(console.error);
```

Comment:
Just confirming I understand this, `useEffect` is used here because there's a network call being made?

Reply:
`useEffect` is just a generic hook for saying "Run this every time any of my dependencies change." So in the case of this one, it'd run `api.audit()` any time `api` changes. Which never happens in practice here, but you can imagine if `audit` had an arugument like pagination, then it would run any time that updated too.

`useEffect` is often paired with network requests since you don't want them running every render, just every time some relevant piece of information changes, but that's not _all_ they're good for. They can also be useful for reacting to changes in component props, querying the DOM, or for binding to events on globals like `window`.